documents/MIGRATION_GUIDE.md

```diff
@@ -0,0 +1,954 @@
+for each topic you wish to unsubscribe from.
+
+#### Example of unsubscribing in the v1 SDK
+
+```java
+// Blocking API.
+client.unsubscribe("my/topic");
+client.unsubscribe("another/topic");
+```
+
+```java
+// Non-blocking API.
+public class MyTopic extends AWSIotTopic {    
+    public MyTopic(String topic, AWSIotQos qos) {
+        super(topic, qos);
+    }
+
+    @Override
+    public void onSuccess() {
+        // Called when unsubscribing succeeds.
+    }
+    
+        @Override
+    public void onFailure() {
+        // Called when unsubscribing fails.
+    }
+}
+
+// Unsubscribe from topic.
+MyTopic myOwnTopic = new MyTopic("my/topic", AWSIotQos.QOS1);
+client.unsubscribe(myOwnTopic);
+```
+
+#### Example of unsubscribing in the v2 SDK
+
+```java
+UnsubscribePacketBuilder unsubBuilder = new UnsubscribePacketBuilder("my/topic");
+client.unsubscribe(unsubBuilder.build()).get(60, TimeUnit.SECONDS);
+```
+
+### Client stop
+
+In the v1 SDK, the `disconnect` method in the `AWSIotMqttClient` class disconnects the client. Once disconnected, the client
+can connect again by calling `connect`.
+
+In the v2 SDK, an MQTT5 client can stop a session by calling the [stop](https://awslabs.github.io/aws-crt-java/software/amazon/awssdk/crt/mqtt5/Mqtt5Client.html#stop(software.amazon.awssdk.crt.mqtt5.packets.DisconnectPacket))
+method. You can provide an optional [DisconnectPacket](https://awslabs.github.io/aws-crt-java/software/amazon/awssdk/crt/mqtt5/packets/DisconnectPacket.html)
+parameter. A closed client can be started again by calling [start](https://awslabs.github.io/aws-crt-java/software/amazon/awssdk/crt/mqtt5/Mqtt5Client.html#start()).
+
+#### Example of disconnecting a client in the v1 SDK
+
+```java
+client.disconnect();
+```
+
+#### Example of disconnecting a client in the v2 SDK
+
+```java
+DisconnectPacketBuilder disconnectBuilder = new DisconnectPacketBuilder();
+disconnectBuilder.withReasonCode(DisconnectPacket.DisconnectReasonCode.NORMAL_DISCONNECTION);
+client.stop(disconnectBuilder.build());
+```
+
+### Client shutdown
+
+The v1 SDK automatically cleans resources allocated by an `AWSIotMqttClient` object on shutdown.
+
+In the v2 SDK, when an MQTT5 client is no longer required, your program **must** close it explicitly via a `close` call.
+
+The v2 SDK `Mqtt5Client` class implements the [AutoCloseable](https://docs.oracle.com/javase/8/docs/api/java/lang/AutoCloseable.html)
+interface, so it is recommended to create `Mqtt5Client` objects in the [try-with-resources](https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html)
+blocks.
+
+#### Example of closing a client in the v2 SDK
+
+```java
+Mqtt5CLient client = builder.build()
+
+// Once fully finished with the Mqtt5Client:
+client.close();
+```
+
+```java
+// client.close() will be called at the end of the try/catch block.
+try (Mqtt5CLient client = builder.build()) {
+    // ...
+} catch (Exception e) {
+}
+
+```
+
+### Reconnects
```

Comment:
the highlighting looks off here, in the generated format they look fine tough

Reply:
Not sure, in my editor the highlighting seems ok. Since the generated doc is also fine, I'll leave it as is.